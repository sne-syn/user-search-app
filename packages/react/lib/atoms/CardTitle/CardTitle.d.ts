import { FC } from "react";
interface CardTitleProps {
    content: string;
}
declare const CardTitle: FC<CardTitleProps>;
export default CardTitle;
