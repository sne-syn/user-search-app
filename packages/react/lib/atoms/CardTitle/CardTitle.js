import React from 'react';

const CardTitle = ({ content }) => {
    return React.createElement("h2", { className: "dse-card-title" }, content);
};

export { CardTitle as default };
//# sourceMappingURL=CardTitle.js.map
