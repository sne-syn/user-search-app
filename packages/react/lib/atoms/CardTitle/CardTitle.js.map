{"version":3,"file":"CardTitle.js","sources":[],"sourcesContent":[],"names":[],"mappings":";;;;;;;;"}