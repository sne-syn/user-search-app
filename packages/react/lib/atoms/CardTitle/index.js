export { default } from './CardTitle.js';
//# sourceMappingURL=index.js.map
