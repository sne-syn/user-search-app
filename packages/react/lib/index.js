export { default as Avatar } from './atoms/Avatar/Avatar.js';
export { default as Button } from './atoms/Button/Button.js';
export { default as CardSubtitle } from './atoms/CardSubtitle/CardSubtitle.js';
export { default as CardTitle } from './atoms/CardTitle/CardTitle.js';
export { default as Icon } from './atoms/Icon/Icon.js';
export { default as Input } from './atoms/Input/Input.js';
export { default as Logo } from './atoms/Logo/Logo.js';
export { default as DateInfo } from './atoms/DateInfo/DateInfo.js';
export { default as DefinitionList } from './atoms/DefinitionList/DefinitionList.js';
export { default as ErrorMessage } from './atoms/ErrorMessage/ErrorMessage.js';
export { default as TextBlock } from './atoms/TextBlock/TextBlock.js';
export { default as ThemeToggler } from './atoms/ThemeToggler/ThemeToggler.js';
export { default as VisuallyHidden } from './atoms/VisuallyHidden/VisuallyHidden.js';
export { default as Wrapper } from './atoms/Wrapper/Wrapper.js';
//# sourceMappingURL=index.js.map
