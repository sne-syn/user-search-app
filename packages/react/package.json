{
  "name": "@user-search-app/react",
  "version": "1.0.0",
  "main": "lib/index.js",
  "license": "MIT",
  "scripts": {
    "build": "rollup -c",
    "dev": "yarn build --watch",
    "test": "jest --verbose",
    "test:watch": "yarn test --watch"
  },
  "devDependencies": {
    "@babel/preset-env": "^7.18.10",
    "@babel/preset-react": "^7.18.6",
    "@babel/preset-typescript": "^7.18.6",
    "@rollup/plugin-image": "^2.1.1",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.3.0",
    "@types/jest": "^28.1.6",
    "@types/react": "^18.0.15",
    "jest": "^28.1.3",
    "jest-environment-jsdom": "^28.1.3",
    "react": "^18.2.0",
    "rollup": "^2.77.0",
    "rollup-plugin-img": "^1.1.0",
    "rollup-plugin-typescript2": "^0.32.1",
    "typescript": "^4.7.4"
  }
}
