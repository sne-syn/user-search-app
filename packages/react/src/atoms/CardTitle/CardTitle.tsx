import React, { FC } from "react";

interface CardTitleProps {
  content: string;
  style?: React.CSSProperties;
}

const CardTitle: FC<CardTitleProps> = ({ content, style }) => {
  return (
    <h2 className="dse-card-title" style={style} data-testid="card-title">
      {content}
    </h2>
  );
};

export default CardTitle;
