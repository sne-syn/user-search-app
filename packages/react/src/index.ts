import Avatar from "./atoms/Avatar";
import Button from "./atoms/Button";
import CardSubtitle from "./atoms/CardSubtitle";
import CardTitle from "./atoms/CardTitle";
import Input from "./atoms/Input";
import ListContent from "./atoms/ListContent";
import ListIcon from "./atoms/ListIcon";
import Logo from "./atoms/Logo";
import DateInfo from "./atoms/DateInfo";
import DefinitionList from "./atoms/DefinitionList";
import ErrorMessage from "./atoms/ErrorMessage";
import TextBlock from "./atoms/TextBlock";
import ThemeToggler from "./atoms/ThemeToggler";
import VisuallyHidden from "./atoms/VisuallyHidden";
import Wrapper from "./atoms/Wrapper";

export {
  Avatar,
  Button,
  CardSubtitle,
  CardTitle,
  Input,
  ListContent,
  ListIcon,
  Logo,
  DateInfo,
  DefinitionList,
  ErrorMessage,
  TextBlock,
  ThemeToggler,
  VisuallyHidden,
  Wrapper,
};
