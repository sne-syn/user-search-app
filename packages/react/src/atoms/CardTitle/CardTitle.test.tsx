import React from "react";
import CardTitle from "./CardTitle";
import { render, screen } from "@testing-library/react";
const CONTENT = "Test CardTitle";

describe("CardTitle", () => {
  test("renders with text content", () => {
    render(<CardTitle content={CONTENT} />);
    expect(screen.getByTestId("card-title")).toHaveTextContent(CONTENT);
    expect(screen.getByTestId("card-title")).toMatchSnapshot();
  });

  test("renders with style provided", () => {
    render(
      <CardTitle content={CONTENT} style={{ backgroundColor: "black" }} />
    );
    expect(screen.getByTestId("card-title")).toHaveStyle(
      "backgroundColor: black"
    );
    expect(screen.getByTestId("card-title")).toMatchSnapshot();
  });
});
