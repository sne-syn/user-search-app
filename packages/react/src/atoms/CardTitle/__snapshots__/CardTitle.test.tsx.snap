// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTitle renders with style provided 1`] = `
<h2
  class="dse-card-title"
  data-testid="card-title"
  style="background-color: black;"
>
  Test CardTitle
</h2>
`;

exports[`CardTitle renders with text content 1`] = `
<h2
  class="dse-card-title"
  data-testid="card-title"
>
  Test CardTitle
</h2>
`;
